jobs: TestRegistryGCPagination failed

jobs.TestRegistryGCPagination [failed](https://tanzanite.cluster.engflow.com/invocations/default/cb776ceb-d010-496c-abe7-fb6410e6eaa3?testReportRun=1&testReportShard=4&testReportAttempt=1#targets-Ly9wa2cvam9iczpqb2JzX3Rlc3Q=) on master @ [5f76d93b2ad8d29c14c9b176753ec33cbf81e3fe](https://github.com/cockroachdb/cockroach/commits/5f76d93b2ad8d29c14c9b176753ec33cbf81e3fe):


```
=== RUN   TestRegistryGCPagination
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestRegistryGCPagination1075318094
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    registry_test.go:319: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/jobs/registry_test.go:319
        	Error:      	Should be zero, but was 12
        	Test:       	TestRegistryGCPagination
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestRegistryGCPagination1075318094
--- FAIL: TestRegistryGCPagination (105.15s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=4</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/jobs @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegistryGCPagination.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34768

--- a/pkg/jobs/registry_test.go
+++ b/pkg/jobs/registry_test.go
@@ -315,7 +315,7 @@ func TestRegistryGCPagination(t *testing.T) {
 	ts := timeutil.Now()
 	require.NoError(t, s.JobRegistry().(*Registry).cleanupOldJobs(ctx, ts.Add(-10*time.Minute)))
 	var count int
-	db.QueryRow(t, `SELECT count(1) FROM system.jobs`).Scan(&count)
+	db.QueryRow(t, `SELECT count(1) FROM system.jobs WHERE status = $1`, StatusCanceled).Scan(&count)
 	require.Zero(t, count)
 }
 

